### DISCUSSION TITLE
Using with Digital Ocean Spaces

### DISCUSSION THREAD

**Discussion by adman9000** [joshembling/image-optimizer#4] [General]
Do we need to do anything special for it to work with DO spaces?  I'm getting an Intervention\Image\Exception\NotReadableException  as it appears to be looking to local storage for the image to optimise rather than to the Spaces storage where it was uploaded to.

Everything works fine when I use local storage, and also fine when I remove the optimise("webp") function and simply save the image to Spaces without optimisation. Is there a config option somewhere to tell the optimise function where to look?

**Comment by adman9000** [comment]
I've discovered that uncommenting line 184 in BaseFileUpload.php
`$originalBinaryFile = $file->get();`
and changing line 191 to 
`$image = InterventionImage::make($originalBinaryFile);`
gets things working ok but unsure whether it has introduced any other bugs. I note this is how it is coded in SpatieMediaLibraryFileUpload.php

**Reply by joshembling** [reply]
Hi @adman9000 

Yes, it was my thinking initially to keep them the same, but the original `FileUpload` plugin has slightly different functionality to `SpatieMediaLibraryFileUpload` and making an intervention image instance based off the binary file breaks explicitly in `FileUpload`. 

You are welcome to make a PR if you have a fully tested solution.

**Reply by azzazkhan** [reply]
@joshembling Same concern for Cloudflare R2, Livewire allows us to use the S3 compatible storage for storing temporary file uploads via signed routes. Will the image optimization work when the image is never stored on local disk?
